xcube/webapi/auth.py

```diff
@@ -199,38 +234,60 @@ def get_id_token(self, require_auth: bool = False) \
                     "e": key["e"]
                 }
                 break
+        if rsa_key is None:
+            raise ServiceAuthError(
+                "Invalid header",
+                log_message="Unable to find appropriate key."
+            )
 
-        if rsa_key:
-            try:
-                id_token = jwt.decode(
-                    access_token,
-                    # TODO: this is stupid: we convert rsa_key to
-                    #  JWT JSON only to produce the public key JSON string
-                    RSAAlgorithm.from_jwk(json.dumps(rsa_key)),
-                    algorithms=auth_config.algorithms,
-                    audience=auth_config.audience,
-                    issuer=auth_config.issuer
-                )
-            except jwt.ExpiredSignatureError:
-                raise ServiceAuthError(
-                    "Token expired",
-                    log_message="Token is expired"
-                )
-            except jwt.InvalidTokenError:
-                raise ServiceAuthError(
-                    "Invalid claims",
-                    log_message="Incorrect claims,"
-                                " please check the audience and issuer"
-                )
-            except Exception:
+        # Now we are ready to decode the access token
+        try:
+            id_token = jwt.decode(
+                access_token,
+                RSAAlgorithm.from_jwk(rsa_key),
+                algorithms=auth_config.algorithms,
+                audience=auth_config.audience,
+                issuer=auth_config.issuer
+            )
+        except jwt.PyJWTError as e:
+            msg = f"Failed to decode access token: {e}"
+            raise ServiceAuthError(msg, log_message=msg) from e
+
+        return id_token
+
+    def get_access_token(self, require_auth: bool = False) -> Optional[str]:
+        """
+        Obtain the request's access token from the Authorization Header.
+        """
+        # noinspection PyUnresolvedReferences
+        auth = self.request.headers.get("Authorization", None)
```

Comment:
I don't like the variable name `auth`. Maybe `auth_header`?

Reply:
I disagree because this is not a header but it is the value of the "Authorization" header. I can rename the variable to `authorization` if this helps you understanding the code, but in this simple case even `a` should be fine.